src/UseCases/ApplyForMembership/MembershipApplicationValidator.php

```diff
@@ -65,9 +53,14 @@ public function validate( ApplyForMembershipRequest $applicationRequest ): Resul
 		$this->validateApplicantContactInfo();
 		$this->validateApplicantDateOfBirth();
 		$this->validateApplicantAddress();
-		if ( $applicationRequest->getPaymentType() === PaymentMethod::DIRECT_DEBIT ) {
-			$this->validateBankData();
-			$this->checkForBlockedIban();
+		// TODO test this
+		if ( $applicationRequest->getPaymentType() == PaymentTypes::DirectDebit ) {
+			$this->paymentValidator->validatePaymentData(
+				$this->request->getPaymentAmountInEuros()->getEuroCents(),
+				$this->request->getPaymentIntervalInMonths(),
+				$this->request->getPaymentType()
+			);
+			// TODO check the violations? more handling?
```

Comment:
Please use the values from `$applicationRequest->getPaymentRequest()` instead.

The payment validator expects a domain-specific validator. Previously, this was done with the `validateFee` method in this class. Now that methods has to be extracted into a class that implements `DomainSpecificPaymentValidator` from the payment domain. The new validator must also validate

* Interval must not be one-time
* Payment type must be in the list of allowed membership payment types (currently a list with only one item, direct debit)